### ISSUE TITLE
Return handler on ShouldBindJSON error

### ISSUE THREAD

**Issue by ericlingit** [eliben/code-for-blog#32]
I'm following your excellent tutorial on writing a Go REST server. [Part 3](https://eli.thegreenplace.net/2021/rest-servers-in-go-part-3-using-a-web-framework/) uses gin framework.

In the gin version of [`createTaskHandler`](https://github.com/eliben/code-for-blog/blob/9fc1e57b137716b8b49850bbc8b1837d1b5d2324/2021/go-rest-servers/gin/gorest-gin.go#L35), only [`c.String(http.StatusBadRequest, err.Error())`](https://github.com/eliben/code-for-blog/blob/9fc1e57b137716b8b49850bbc8b1837d1b5d2324/2021/go-rest-servers/gin/gorest-gin.go#L44) is called if `err` isn't `nil`. Should there be a `return` statement at the end of the error handling block?

In other words:

```go
func (ts *taskServer) createTaskHandler(c *gin.Context) {
	type RequestTask struct {
		Text string    `json:"text"`
		Tags []string  `json:"tags"`
		Due  time.Time `json:"due"`
	}

	var rt RequestTask
	if err := c.ShouldBindJSON(&rt); err != nil {
		c.String(http.StatusBadRequest, err.Error())
		//
		// Should there be a `return` statement here?
		//
	}

	id := ts.store.CreateTask(rt.Text, rt.Tags, rt.Due)
	c.JSON(http.StatusOK, gin.H{"Id": id})
}
```

**Comment by eliben** [comment]
Thanks for the comment! This should be fixed in the code now (and the blog page will be redeployed with the fix at some point too)
